cpp/src/graphar/high-level/graph_reader.cc

```diff
@@ -94,6 +100,290 @@ Vertex::Vertex(IdType id,
+    std::vector<std::string> filter_labels) const {
+  std::vector<int> indices;
+  const int TOT_ROWS_NUM = vertex_num_;
+  const int CHUNK_SIZE = vertex_info_->GetChunkSize();
+
+  std::vector<int> tested_label_ids;
+  for (const auto& filter_label : filter_labels) {
+    auto it = std::find(labels_.begin(), labels_.end(), filter_label);
+    if (it != labels_.end()) {
+      tested_label_ids.push_back(std::distance(labels_.begin(), it));
+    }
+  }
+  int total_count = 0;
+  int row_num;
+  std::vector<std::shared_ptr<Expression>> filters;
+  std::shared_ptr<Expression> combined_filter = nullptr;
+
+  for (const auto& label : filter_labels) {
+    filters.emplace_back(
+        graphar::_Equal(graphar::_Property(label), graphar::_Literal(true)));
+  }
+
+  for (const auto& filter : filters) {
+    if (!combined_filter) {
+      combined_filter = graphar::_And(filter, filter);
+    } else {
+      combined_filter = graphar::_And(combined_filter, filter);
+    }
+  }
+
+  auto maybe_filter_reader = graphar::VertexPropertyArrowChunkReader::Make(
+      vertex_info_, labels_, prefix_, {});
+  auto filter_reader = maybe_filter_reader.value();
+  filter_reader->Filter(combined_filter);
+  for (int chunk_idx = 0; chunk_idx * CHUNK_SIZE < TOT_ROWS_NUM; ++chunk_idx) {
+    auto filter_result = filter_reader->GetLabelChunk();
+    auto filter_table = filter_result.value();
+    total_count += filter_table->num_rows();
+    filter_reader->next_chunk();
+  }
+  // std::cout << "Total valid count: " << total_count << std::endl;
+  std::vector<int64_t> indices64;
+
+  for (int value : indices) {
+    indices64.push_back(static_cast<int64_t>(value));
+  }
+
+  return indices64;
+}
+
+Result<std::shared_ptr<VerticesCollection>>
+VerticesCollection::verticesWithLabel(
+    const std::string& filter_label,
+    const std::shared_ptr<GraphInfo>& graph_info, const std::string& type) {
+  auto prefix = graph_info->GetPrefix();
+  auto vertex_info = graph_info->GetVertexInfo(type);
+  auto labels = vertex_info->GetLabels();
+  auto vertices_collection =
+      std::make_shared<VerticesCollection>(vertex_info, prefix);
+  vertices_collection->filtered_ids_ =
+      vertices_collection->filter({filter_label}).value();
+  vertices_collection->is_filtered_ = true;
+  return vertices_collection;
+}
+
+Result<std::shared_ptr<VerticesCollection>>
+VerticesCollection::verticesWithLabelbyAcero(
+    const std::string& filter_label,
+    const std::shared_ptr<GraphInfo>& graph_info, const std::string& type) {
+  auto prefix = graph_info->GetPrefix();
+  auto vertex_info = graph_info->GetVertexInfo(type);
+  auto labels = vertex_info->GetLabels();
+  auto vertices_collection =
+      std::make_shared<VerticesCollection>(vertex_info, prefix);
+  vertices_collection->filtered_ids_ =
+      vertices_collection->filter_by_acero({filter_label}).value();
+  vertices_collection->is_filtered_ = true;
+  return vertices_collection;
+}
+
+Result<std::shared_ptr<VerticesCollection>>
```

Comment:
Single label filtering API, input is VerticesCollection ( a result from another filtering)